libretroshare/src/ft/ftserver.cc

```diff
@@ -2101,3 +2206,144 @@ RsFileItem::RsFileItem(RsFileItemType subtype) :
 void RsFileSearchRequestItem::clear() { queryString.clear(); }
 
 void RsFileSearchResultItem::clear() { mResults.clear(); }
+
+
+std::error_condition RsFilesErrorCategory::default_error_condition(int ev)
+const noexcept
+{
+	switch(static_cast<RsFilesErrorNum>(ev))
+	{
+	case RsFilesErrorNum::FILES_HANDLE_NOT_FOUND:
+		return std::errc::invalid_argument;
+	default:
+		return std::error_condition(ev, *this);
+	}
+}
+
+std::error_condition ftServer::dirDetailsToLink(
```

Comment:
It encode fields properly following the RFC standard, so special characters won't break it. Also in case parsing fails it just return an error and no more fuss ;)
BTW the relevant data is encoded in base64url format, the URL is just an disguise to encrust the data into in a way that suggests other application to make it clickable, it is important to do it at libretroshare level so all application using libretroshare have compatible links formats, without becoming crazy on making different URL libraries to behave similarly (they usually don't except for http[s] links that are the most common and tested). We already do this for channels, forums etc